### PR TITLE
use USER env and create svc with yaml instead of expose + other changes

### PR DESCRIPTION
**Pull request by splattner**

### CODE REVIEW

File: content/en/docs/additional-concepts/sidecar-containers/_index.md [hunk 1]
```diff
@@ -25,28 +25,23 @@ In a sidecar pattern, the functionality of the main container is extended or enh
 In {{<link "persistent-storage">}} you created a MariaDB deployment. In this task you are going to add the [Prometheus MySQL exporter](https://github.com/prometheus/mysqld_exporter) to it.
 
 {{% onlyWhenNot openshift %}}
```
Comment (splattner) [inline content/en/docs/additional-concepts/sidecar-containers/_index.md:27]: shall I change this also for OpenShift?

Comment (bliemli) [reply content/en/docs/additional-concepts/sidecar-containers/_index.md:27]: What exactly?

**Comment by bliemli** [comment]
Why use `$USER` if you mean the namespace? The username doesn't necessarily have to be the namespace.

File: content/en/docs/additional-concepts/sidecar-containers/_index.md (hunk 1 above)
Comment (bliemli) [reply content/en/docs/additional-concepts/sidecar-containers/_index.md:27]: Ah, in the overview it didn't show the following changes. We can also change this in OpenShift, that's ok, thanks.

**Review by bliemli — COMMENTED** [review]
I couldn't yet review all of it and will certainly have to test all the changes in more detail, but these are my first suggestions.

**Comment by splattner** [comment]
> Why use `$USER` if you mean the namespace? The username doesn't necessarily have to be the namespace.
> 
> Or even better, use placeholders that can be set via URL parameters instead of letting the user set env vars. The reason for this is, depending on the training, we have different naming conventions, especially if we are on customer-provided infrastructure.

I think @g1raffi already used thi in the helm training, so I reused it from there. Also `$USER` is set in the webshell and so the trainees can simply copy/paste it.
We should use the same scheme in all training. Let's align on this in a S1 Stream meeting or update day

File: content/en/docs/exposing-a-service/_index.md [hunk 2]
```diff
@@ -26,10 +26,15 @@ You can also use Ingress to expose your Service. Ingress is not a Service type,
...
-We're going to do this with the command `{{% param cliToolName %}} expose`:
+
+To create the Service add a new file `svc-web-go.yaml` with the following content:
+
+{{< readfile file="/content/en/docs/exposing-a-service/svc-web-go.yaml" code="true" lang="yaml" >}}
+
+And then apply the file with:
 
 ```bash
-{{% param cliToolName %}} expose deployment example-web-go --type=ClusterIP --name=example-web-go --port=5000 --target-port=5000 --namespace <namespace>
```
[the remark this answers is not in the document]
Comment (splattner) [reply content/en/docs/exposing-a-service/_index.md:32]: If I remember correctly we once discussed, that we wan't to get rid of these imperative commands. Not sure if you were involved in this discussion though. In my opinion the `--dry-run=client -o yaml` approach is the nicer one to create resources for the gitops approach. But I guess for `expose` there is not relly an equivalent.
Maybee also something to quickly discuss/align in a S1 Stream meeting.

**Comment by schlapzz** [comment]
As Seba already mentioned, we're make heavy use of this Env Var in our Trainings ($LAB_USER in AMM Techlab, later renamed in the Helm & ArgoCD Techlab from $STUDENT to $USER, see [here](https://github.com/acend/argocd-basics-training/issues/243 and later [here](https://github.com/acend/helm-basics-training/pull/395))

The naming of the var is not that important for me.  But regarding the new Training Cluster setup it would be nice if we use the same Var Name for all trainings 💙

**Comment by bliemli** [comment]
> As Seba already mentioned, we're make heavy use of this Env Var in our Trainings ($LAB_USER in AMM Techlab, later renamed in the Helm & ArgoCD Techlab from $STUDENT to $USER, see [here](https://github.com/acend/argocd-basics-training/issues/243) and later [here](https://github.com/acend/helm-basics-training/pull/395)
> 
> The naming of the var is not that important for me. But regarding the new Training Cluster setup it would be nice if we use the same Var Name for all trainings blue_heart

Still, the best option imho would be to use a placeholder as described [here](https://github.com/acend/docsy-plus#dynamically-replace-content). This doesn't require setting an environment variable on the terminal and is widely used in the Kubernetes Basics training.

**Comment by splattner** [comment]
I replaced `<namespace>` with `$USER` using the placeholder feature and I also introduced a placeholder for appdomain.

File: content/en/docs/exposing-a-service/_index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (splattner) [reply content/en/docs/exposing-a-service/_index.md:32]: I added the `expose` command again as a second option.

File: content/en/docs/additional-concepts/sidecar-containers/_index.md (hunk 1 above)
Comment (splattner) [reply content/en/docs/additional-concepts/sidecar-containers/_index.md:27]: And this is also changed in the oc variants

**Review by bliemli — APPROVED** [review]

**Pull request merged**